sdks/python/apache_beam/io/vcfio.py

```diff
@@ -0,0 +1,440 @@
+      return self.end < other.end
+
+    self_vars = vars(self)
+    other_vars = vars(other)
+    for key in sorted(self_vars):
+      if self_vars[key] != other_vars[key]:
+        return self_vars[key] < other_vars[key]
+
+    return False
+
+  def __le__(self, other):
+    if not isinstance(other, Variant):
+      return NotImplemented
+
+    return self < other or self == other
+
+  def __ne__(self, other):
+    if not isinstance(other, Variant):
+      return NotImplemented
+
+    return not self == other
+
+  def __gt__(self, other):
+    if not isinstance(other, Variant):
+      return NotImplemented
+
+    return other < self
+
+  def __ge__(self, other):
+    if not isinstance(other, Variant):
+      return NotImplemented
+
+    return other <= self
+
+
+class VariantCall(object):
+  """A class to store info about a variant call.
+
+  A call represents the determination of genotype with respect to a particular
+  variant. It may include associated information such as quality and phasing.
+  """
+
+  def __init__(self, name=None, genotype=None, phaseset=None, info=None):
+    """Initialize the :class:`VariantCall` object.
+
+    Args:
+      name (str): The name of the call.
+      genotype (List[int]): The genotype of this variant call as specified by
+        the VCF schema. The values are either `0` representing the reference,
+        or a 1-based index into alternate bases. Ordering is only important if
+        `phaseset` is present. If a genotype is not called (that is, a `.` is
+        present in the GT string), -1 is used
+      phaseset (str): If this field is present, this variant call's genotype
+        ordering implies the phase of the bases and is consistent with any other
+        variant calls in the same reference sequence which have the same
+        phaseset value. If the genotype data was phased but no phase set was
+        specified, this field will be set to `*`.
+      info (dict): A map of additional variant call information. The key is
+        specified in the VCF record and the type of the value is specified by
+        the VCF header FORMAT.
+    """
+    self.name = name
+    self.genotype = genotype or []
+    self.phaseset = phaseset
+    self.info = info or {}
+
+  def __eq__(self, other):
+    return (self.name == other.name and
+            self.genotype == other.genotype and
+            self.phaseset == other.phaseset and
+            self.info == other.info)
+
+  def __repr__(self):
+    return ', '.join(
+        [str(s) for s in [self.name, self.genotype, self.phaseset, self.info]])
+
+
+class _VcfSource(filebasedsource.FileBasedSource):
+  """A source for reading VCF files.
+
+  Parses VCF files (version 4) using PyVCF library. If file_pattern specifies
+  multiple files, then the header from each file is used separately to parse
+  the content. However, the output will be a uniform PCollection of
+  :class:`Variant` objects.
+  """
+
+  DEFAULT_VCF_READ_BUFFER_SIZE = 65536  # 64kB
```

Comment:
<!--thread_id:cc_148154835_t; commit:e00f526d58e10e80c9f0f7b77f2f5d2bd51d9df2; resolved:1-->
<!--section:context-quote-->
> **mhsaul** wrote:
> I think the text source default is 8KB. We increased it to 64KB since VCF lines tend to be longer. A single record with about 1000 samples is 30KB, so we figured 64KB would be sufficient.

<!--section:body-->
Note that this is different from the buffer of file-system implementation. (for example, 16MB for GCS).

Reply:
@chamikaramj what buffer size would you suggest for the reader? I am a bit unsure with how this value should be chosen.